src/components/artwork/fillwidth.tsx

```diff
@@ -0,0 +1,147 @@
+import * as React from "react"
+import * as Relay from "react-relay"
+
+import sizeMe from "react-sizeme"
+import styled from "styled-components"
+import Artwork from "./fillwidth_item"
+
+import { find, map, max, reduce } from "lodash"
+
+interface RelayProps {
+  artworks: {
+    edges: Array<{
+      node: any,
+    } | null> | null,
+  },
+}
+
+interface Props extends RelayProps, React.HTMLAttributes<Fillwidth> {
+  targetHeight?: number
+  gutter?: number
+  size?: any
+}
+
+/**
+ * Scales an image object proportionally based on a direction (either -1 or 1)
+ * @param img a dimension object that references an artwork image
+ * @param dir the direction we need to scale an image, either -1 or 1
+ */
+const resizeHeight = (img, dir) => {
+  img.width += (img.width / img.height) * dir
+  img.height += dir
+}
+
+export class Fillwidth extends React.Component<Props, null> {
+  public static defaultProps: Partial<Props>
+
+  totalWhitespace() {
+    const artworks = this.props.artworks.edges
+    return (artworks.length - 1) * this.props.gutter
+  }
+
+  widthDiff(dimensions) {
+    const currentWidth = reduce(dimensions, (sum, img) => {
+      return sum + img.width
+    }, 0)
+    return this.props.size.width - currentWidth - this.totalWhitespace()
+  }
+
+  getDimensions(containerWidth) {
+    const artworks = this.props.artworks.edges
+
+    // Get initial dimensions based on the targetHeight
+    let dimensions = artworks.map(artwork => {
+      return {
+        __id: artwork.node.__id,
+        width: this.props.targetHeight * artwork.node.image.aspect_ratio,
+        height: this.props.targetHeight,
+      }
+    })
+
+    // If the total width difference is less than 0, it is larger than the container
+    // so we need to scale down. If not, scale up.
+    let dir = this.widthDiff(dimensions) < 0 ? -1 : 1
+
+    // Keep looping until we get an acceptable width difference
+    const list = new Array(999)
+
+    for (let i of list) { // eslint-disable-line no-unused-vars
```

Comment:
Is the intention here to just loop indefinitely until `break` is called? If so, something like the following will get rid of the unused array and unused variable:

```js
let proceed = true
while (proceed) {
  // ...
}
```